src/main/scala/org/apache/mesos/chronos/scheduler/jobs/Schedule.scala

```diff
@@ -1,44 +1,162 @@
+ * @param invocationTime the time this schedule needs to be invoked (UTC)
+ * @param originTime the root time that the ISO 8601 schedule was started (UTC)
+ * @param offset the number of executions after the origin that have completed
+ * @param recurrences the number of remaining executions, or -1 if this is an infinite schedule
+ * @param period the time between executions
+ */
+case class Iso8601Schedule(@JsonProperty schedule: String,
+                           @JsonProperty scheduleTimeZone: String,
+                           @JsonProperty invocationTime: DateTime,
+                           @JsonProperty originTime: DateTime,
+                           @JsonProperty offset: Long = 0,
+                           @JsonProperty recurrences: Option[Long] = None,
+                           @JsonProperty period: Period) extends Schedule {
+  type SType = Iso8601Schedule
 
-  def next: Option[Schedule] = {
+  val scheduleType = Iso8601Type
+
+  override def next: Option[Iso8601Schedule] = {
     if (recurrences.exists(_ == 0)) None
     else {
       val nextOffset = offset + 1
-      val nextInvocationTime = Schedule.addPeriods(originTime, period, nextOffset.asInstanceOf[Int] /* jodatime doesn't support longs in their plus methods. What to do about overflow? */)
+      val nextInvocationTime = Schedules.addPeriods(originTime, period, nextOffset.asInstanceOf[Int] /* jodatime doesn't support longs in their plus methods. What to do about overflow? */)
 
-      Some(Schedule(schedule, scheduleTimeZone, nextInvocationTime, originTime, nextOffset, recurrences.map(_ - 1), period))
+      Some(Iso8601Schedule(schedule, scheduleTimeZone, nextInvocationTime, originTime, nextOffset, recurrences.map(_ - 1), period))
     }
   }
 
-  def toZeroOffsetISO8601Representation: String = {
+  def toStringRepresentation: String = {
     Iso8601Expressions.create(recurrences.getOrElse(-1), invocationTime, period)
   }
 }
 
-object Schedule {
-  def parse(scheduleStr: String, timeZoneStr: String = ""): Option[Schedule] = {
+/**
+ * A Unix Cron schedule
+ *
+ * @param schedule the text representation of this schedule
+ * @param scheduleTimeZone the time zone the schedule should be executed from
+ * @param invocationTime the time to execute this schedule (UTC)
+ * @param lastExecutionTime the last time this schedule was run (UTC)
+ * @param cron the parsed Cron object that is used to perform execution time planning
+ */
+case class CronSchedule(@JsonProperty schedule: String,
+                        @JsonProperty scheduleTimeZone: String,
+                        @JsonProperty invocationTime: DateTime,
+                        @JsonProperty lastExecutionTime: DateTime,
+                         cron: Cron) extends Schedule {
+  type SType = CronSchedule
+  val recurrences = None
+  val scheduleType = CronType
+
+  lazy val next: Option[CronSchedule] = {
+    val tz = if(scheduleTimeZone != "") DateTimeZone.forID(scheduleTimeZone) else DateTimeZone.UTC
```

Comment:
Code style: add a space after `if`. Also use `isEmpty` instead of comparing with `""`. And I would add `.trim` before the `isEmpty` test just in case it is all whitespace.